modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/externalize/RelJsonReader.java

```diff
@@ -84,6 +86,21 @@ public static <T extends RelNode> T fromJson(BaseQueryContext ctx, String json)
         return (T) reader.read(json);
     }
 
+    /** Creates a {@link RexNode} from the given json. */
+    public static RexNode fromExprJson(byte[] json) {
```

Comment:
why `byte[]` ?

Reply:
`fromExprJson` uses bytes because `toJsonExpr` converts expressions to bytes[].
And `toJsonExpr` returns an array of bytes, because serialised expressions are then immediately written to output stream.

An alternative approach would be to convert expression into a string and then use methods from data stream than convert string into bytes.